### ISSUE TITLE
Unable to build from source for macOS

### ISSUE THREAD

**Issue by georgesittas** [kchasialis/QtMips-Di#1]
What the title says. 

I cloned the repository locally and followed the steps described in "Building from source on macOS" (*)
but I can't seem to find the `build-macos.sh` script anywhere in order to run the build. If it helps, I am
using QMake 3.1 and Qt version 6.2.1  -- what homebrew installed for me by default.

Can you help me out? Thanks in advance!

(*) Apart from the Xcode step, because it seemed unimportant. Let me know if this assumption was wrong and I'll see to it.

**Comment by kchasialis** [comment]
Hello.

This is an extended version of the original QtMips. I added a branch predictor, stall-on-branch option, branch resolution on EX stage and L2 cache. 
If you want the original QtMips you can use this repository: https://github.com/cvut/QtMips <https://github.com/cvut/QtMips>

If, however, you want my version, I regret to inform you that currently macOS is not supported. I need to update the README and documentation.

Best,
Kostas.

**Comment by kchasialis** [comment]
You can, however, build on macOS using cmake and ninja.

Instructions:
install cmake for macOS (brew install cmake)
install ninja for macOS (brew install ninja)
git clone https://github.com/kchasialis/QtMips-Di <https://github.com/kchasialis/QtMips-Di>
cd QtMips-Di
mkdir build
cd build
cmake .. -G Ninja
ninja

and if the build succeeds you can run QtMips by typing ./target/qtmips_gui on the terminal from the build directory.

Please ask me any questions if you encounter any problems! I will be happy to help.

Best,
Kostas.

**Comment by georgesittas** [comment]
> This is an extended version of the original QtMips. I added a branch predictor, stall-on-branch option, branch resolution on EX stage and L2 cache.

I have already played around a little with your version of the simulator in a Windows environment, and the additions seemed really nice. Congratulations for your work!

> You can, however, build on macOS using cmake and ninja [...].

I managed to get the build to succeed somehow, but it was not trivial because the compiler had trouble locating ELF-related directories and headers, even after I installed libelf through Homebrew. I ended up manually changing some of the `#include` directives in the project, replacing <...> paths with "..." paths that lead to the corresponding ELF files in my system. Seems like this solved the problem.

Thank you for your help! :-)
